internal/manifests/collector/configmap.go

```diff
@@ -18,19 +18,30 @@ func ConfigMap(params manifests.Params) (*corev1.ConfigMap, error) {
 
 	replacedConf, err := ReplaceConfig(params.OtelCol)
 	if err != nil {
-		params.Log.V(2).Info("failed to update prometheus config to use sharded targets: ", "err", err)
+		params.Log.V(2).Info("failed to update config: ", "err", err)
 		return nil, err
 	}
 
+	sourceDataMap := map[string]string{
+		"cwagentconfig.json": replacedConf,
```

Comment:
nit: Is there a reason we aren't using `params.Config.CollectorConfigMapEntry()` instead of the hardcoded "cwagentconfig.json" string?

Reply:
If we stick to this convention, we should ideally be consistent across all `configmap.go` files under `internal/manifests`, right?

I know upstream explicitly writes the string out: https://github.com/open-telemetry/opentelemetry-operator/blob/main/internal/manifests/collector/configmap.go#L54.

Reply:
Its to avoid hard-coding strings. Just because upstream does it, doesnt mean the operator should as well.

Reply:
Makes sense.